src/constant_time.rs

```diff
@@ -22,9 +22,7 @@ use {c, error};
 /// `a` and `b`.
 pub fn verify_slices_are_equal(a: &[u8], b: &[u8])
                                -> Result<(), error::Unspecified> {
-    if a.len() != b.len() {
-        return Err(error::Unspecified);
-    }
+    check!(a.len() == b.len());
     let result = unsafe { GFp_memcmp(a.as_ptr(), b.as_ptr(), a.len()) };
     match result {
```

Comment:
It definitely could (I was trying to be conservative, pending more general discussion).